### PR TITLE
New: Multicomponent compressible model

### PR DESCRIPTION
**Pull request by bobmyhill**
This PR implements the simplest geologically-reasonable multicomponent compressible equation of state (at least, the simplest I can come up with). It is based on the Murnaghan equation of state with a constant specific Cv and a thermal pressure term that is linear in T.

All composite thermodynamic properties are averaged self-consistently.

Derived from the new split equation of state/material model (#3075, now merged into master).

### CODE REVIEW

**Comment by bobmyhill** [comment]
Hi @gassmoeller ... here's the potential alternative for the simple compressible model. As far as I can see, it behaves pretty well over a wide range of temperatures and pressures - not as well as the more flexible modified Tait, but fine for most geodynamic purposes.

File: source/material_model/equation_of_state/multicomponent_compressible.cc [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+            out.densities[c] = reference_densities[c]*std::pow(f, 1./isothermal_bulk_modulus_pressure_derivatives[c]);
+            out.thermal_expansion_coefficients[c] = reference_thermal_expansivities[c] / f;
+            out.specific_heat_capacities[c] = (isochoric_specific_heats[c] +
+                                               (temperature*reference_thermal_expansivities[c] *
+                                                ak * std::pow(f, -1.-(1./isothermal_bulk_modulus_pressure_derivatives[c]))
+                                                / reference_densities[c]));
+            out.compressibilities[c] = reference_isothermal_compressibilities[c]/f;
+            out.entropy_derivative_pressure[c] = -out.thermal_expansion_coefficients[c] / out.densities[c];
+            out.entropy_derivative_temperature[c] = out.specific_heat_capacities[c] / temperature;
```
Comment (gassmoeller) [inline source/material_model/equation_of_state/multicomponent_compressible.cc]: We usually use these derivatives for latent heat generation at phase boundaries only, but the current implementation looks like it always produces derivatives. Let's have a chat about this, I feel like I mislead you, when you asked if these derivatives were isothermal/isobaric.

File: source/material_model/equation_of_state/multicomponent_compressible.cc [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                                                                                list_of_composition_names,
+                                                                                has_background_field,
+                                                                                "Reference thermal expansivities");
+
+        isochoric_specific_heats = Utilities::parse_map_to_double_array (prm.get("Isochoric specific heats"),
+                                                                         list_of_composition_names,
+                                                                         has_background_field,
+                                                                         "Isochoric specific heats");
+      }
+
```
Comment (gassmoeller) [inline source/material_model/equation_of_state/multicomponent_compressible.cc]: remove this empty line

File: source/material_model/multicomponent_compressible.cc [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    {
+      EquationOfStateOutputs<dim> eos_outputs (this->n_compositional_fields()+1);
+
+      for (unsigned int i=0; i < in.temperature.size(); ++i)
+        {
+          const std::vector<double> reference_volume_fractions = MaterialUtilities::compute_volume_fractions(in.composition[i]); // these are the reference (uncompressed) volume fractions - ASPECT does not currently compute the changes in volume fraction with pressure and temperature
+
+          equation_of_state.evaluate(in, i, eos_outputs);
+
+          unsigned int n_fields = reference_volume_fractions.size();
```
Comment (gassmoeller) [inline source/material_model/multicomponent_compressible.cc]:
```suggestion
          const unsigned int n_fields = reference_volume_fractions.size();
```

File: tests/compressible_forsterite_box.prm [hunk 4]
```diff
@@ -0,0 +1,88 @@
+# This test ensures that the Modified Tait
```
Comment (gassmoeller) [inline tests/compressible_forsterite_box.prm]: multicomponent compressible

File: source/material_model/equation_of_state/multicomponent_compressible.cc (hunk 1 above)
Comment (bobmyhill) [reply source/material_model/equation_of_state/multicomponent_compressible.cc]: Ok, corrected! Thanks for catching this.

**Comment by gassmoeller** [comment]
Is this ready?

**Comment by bobmyhill** [comment]
Yes! :)

**Pull request merged**